### PR TITLE
Bug 1189390 - fix project selector so it gets re-rendered after project list changes

### PR DESCRIPTION
**Pull request by jwforres**

### CODE REVIEW

**Comment by jwforres** [comment]
@liggitt review

File: assets/app/scripts/directives/nav.js [hunk 1]
```diff
@@ -27,26 +27,42 @@ angular.module('openshiftConsole')
...
-              var currentURL = $location.url();
-              var currProjRegex = /\/project\/[^\/]+/;
-              var currProjPrefix = currProjRegex.exec(currentURL);
-              var newURL = currentURL.replace(currProjPrefix, "/project/" + encodeURIComponent(newProject));
-              $scope.$apply(function() {
-                $location.url(newURL);
-              });
-            });
+        var select = $('.selectpicker', element);
+        $each($scope.projects, function(name, project) {
```
Comment (liggitt) [inline assets/app/scripts/directives/nav.js]: make this a function we can call on load and on project upate

File: assets/app/views/_project-nav.html [hunk 2]
```diff
@@ -4,7 +4,6 @@
       <label class="control-label" for="boostrapSelect">Projects</label>
       <!-- TODO add multiple attribute back to the select tag when we support selecting multiple -->
       <select class="selectpicker form-control" data-selected-text-format="count>3" id="boostrapSelect">
-        <option ng-selected="selected == proj.metadata.name" ng-repeat="proj in projects" value="{{proj.metadata.name}}">{{proj.displayName || proj.metadata.name}}</option>
         <!-- TODO add back in when we support create project
```
Comment (liggitt) [inline assets/app/views/_project-nav.html]: remove TODO, move to where we build the options now

**Comment by liggitt** [comment]
LGTM after comments are addressed

**Comment by jwforres** [comment]
[merge]

**Comment by jwforres** [comment]
unrelated [merge]

**Pull request merged**